src/handlers/dns/DnsStreamHandler.cpp

```diff
@@ -333,6 +345,45 @@ void DnsStreamHandler::info_json(json &j) const
 {
     common_info_json(j);
     j[schema_key()]["xact"]["open"] = _metrics->num_open_transactions();
+    j[schema_key()]["predicate"]["enabled"] = _using_predicate_signals;
+}
+
+inline void DnsStreamHandler::_register_predicate_filter(std::string f_key, std::string f_value)
+{
+    if (!_udp_qname_predicate_signal && !_udp_qname_predicate_signal) {
+        // all DnsStreamHandler race to install this predicate, which is only installed once and called once per udp event
+        // it's job is to return the predicate "jump key" to call matching signals
+        _udp_qname_predicate = [this](pcpp::Packet &payload, PacketDirection dir, pcpp::ProtocolType l3, uint32_t flowkey, timespec stamp) -> std::string {
```

Comment:
This became specific for qname. The way I thought about `_register_predicate_filter` would work for different type of filters (Qname2, Rcode, etc), for each filter type we are going to have one of this Lambda;;

Reply:
cool, great